src/states/node.rs

```diff
@@ -482,22 +482,31 @@ impl Node {
                 } else if self.tunnels.has_clients(src, dst) {
                     self.send_or_drop(&dst, bytes, content.priority());
                     Ok(())
-                } else if self.tunnels.accept_clients(src, dst) {
-                    debug!("{:?} agreed to act as tunnel_node for {:?} - {:?}",
-                           self,
-                           src,
-                           dst);
-                    self.send_direct_message(dst, DirectMessage::TunnelSuccess(src));
-                    self.send_or_drop(&dst, bytes, content.priority());
-                    Ok(())
+                } else if self.peer_mgr.can_tunnel_for(&src, &dst) {
```

Comment:
If you move the `!self.peer_mgr.can_tunnel_for(&src, &dst)` (negated) part here, you can do without the nested `if`s,
i.e. continue with `else if self.tunnels.accept_clients(src, dst)` below. That might be a bit easier to read.